File: clients/DotNET/Zbus.Examples/Rpc/RpcProcessorExample.cs
using Api.Example;
using System;
using Zbus.Mq;
using Zbus.Rpc;

namespace Zbus.Examples
{
    class RpcProcessorExample
    { 
        static void Main(string[] args)
        { 
            RpcProcessor p = new RpcProcessor();
            p.AddModule<MyService>(); //Simple?


            Broker broker = new Broker("localhost:15555"); //Capable of HA failover, test it!
            Consumer c = new Consumer(broker, "MyRpc");
            c.MessageReceived += p.MessageHandler; //Set processor as message handler
            c.Start();

            Console.WriteLine("done");
            Console.ReadKey();
        }
    }
}

File: src/test/java/io/zbus/examples/rpc/RpcProcessorExample.java
package io.zbus.examples.rpc;

import io.zbus.examples.rpc.api.InterfaceExampleImpl;
import io.zbus.mq.Broker;
import io.zbus.mq.Consumer;
import io.zbus.mq.ConsumerConfig;
import io.zbus.rpc.RpcProcessor;

public class RpcProcessorExample {

	@SuppressWarnings("resource")
	public static void main(String[] args) throws Exception {   
		
		RpcProcessor processor = new RpcProcessor();
		processor.addModule(new InterfaceExampleImpl());  
		
		
		Broker broker = new Broker("localhost:15555");   
		ConsumerConfig config = new ConsumerConfig();
		config.setBroker(broker);
		config.setTopic("MyRpc");
		config.setMessageHandler(processor);   
		
		Consumer consumer = new Consumer(config); 
		
		consumer.start(); 
	} 
}
